client/onboarding/index.tsx

```diff
@@ -77,7 +77,8 @@ const initialData = wcpaySettings.onboardingFlowState?.data ?? {
 
 const OnboardingPage: React.FC = () => {
 	useEffect( () => {
-		trackStarted();
+		const urlParams = new URLSearchParams( window.location.search );
+		trackStarted( urlParams.get( 'source' ) );
```

Comment:
should we sanitize it here or maybe in the method itself? Since it's gotten from the URL and sent to the tracking service

Reply:
@ismaeldcom what is the best way to sanitize strings in TS?

Reply:
@ismaeldcom @dmallory42 what is the best way to sanitize strings in TS? Is there a common way how it's done in our client? I quickly search but didn't find one

Reply:
I don't think we have a sanitize method for strings, [we have one for HTML based on dompurify](https://github.com/Automattic/woocommerce-payments/blob/2d16b990b9e2d0c78483c830f7ae4d5e044946bb/client/utils/sanitize.ts#L7).

For this case probably a simple regex like `/[^\w-]/g` should be enough.

Reply:
fixed in https://github.com/Automattic/woocommerce-payments/pull/8165/commits/454b8a3c38a1ab3bb7babc6db0f69920b2a5e6f5, thanks!